src/Task.ts

```diff
@@ -82,4 +83,8 @@ export function empty<A>(): Task<A> {
   return never as Task<A>
 }
 
+export function fromPromise<L, A>(f: Lazy<Promise<A>>, onrejected: (reason: any) => L): Task<Either<L, A>> {
+  return new Task(() => f().then(a => right<L, A>(a), reason => left<L, A>(onrejected(reason))))
+}
+
```

Comment:
It does feel slightly inconsistent to have `Task.fromPromise` and `either.tryCatch`, which are async and sync versions of the same idea, but with different naming. 🤔